Update scheduled.js

unbeschwert opened the pull request:
Added changes to display link of the submissions when they exists.

unbeschwert committed:
Update scheduled.js

Added changes to display link of the submissions when they exists.

hmadison commented on src/scheduled.js:
```diff
@@ -27,7 +27,13 @@ const readLobstersRss = (event) => {
+          event.reply(`${item.title} [${item.categories.join(' ')}] (${username}) ${item.guid}`);          
+        }
+        else {
+          event.reply(`${item.title} {${item.link}} [${item.categories.join(' ')}] (${username}) ${item.guid}`);
```
Can we check the length of the message before we send it? I feel like adding the URL can create a message that's too long for IRC.

unbeschwert replied:
Hi @hmadison ,

Is the check really necessary ? Recently I got a list of all URLs posted to lobsters from @pushcx  and I created a histogram of these URL lengths which you can find below. Most of the URLs lengths are in range of 30 to 80 characters which seems to be fine IMHO. 

![Figure_1](https://user-images.githubusercontent.com/56976873/120627737-4054b900-c464-11eb-812a-fe048ef79369.png)

hmadison replied:
Pulling the maximum lengths from the schema.rb, we have an upper bound of 707 characters. IRC's max length is 510.

unbeschwert replied:
Ah okay yes. In that case we need to add checks for length. I'll make the necessary changes. Please do make sure to run some tests on them since I am not familiar with javascript

Edit: @hmadison  I just added a new commit that checks if length of reply is greater than 510. I hope this is correct. If it needs some change, please let me know :)

One other idea I had in mind was to use some sort of URL shortnening service. Although they all required paid plans and seemed a bit too much. If its interests you, I can also look it up :)

winks replied:
git.io should work if desired - https://github.blog/2011-11-10-git-io-github-url-shortener/

hmadison replied:
@winks `git.io` is for GitHub controlled properties.

unbeschwert committed:
Update scheduled.js

Added check for length of the reply. Reply string should not be > 510 characters according to IRC spec.

hmadison commented:
Hey @unbeschwert, would it be possible to get the test added we discussed on IRC so I can merge this?
